doc/man/man3/RunGui.3
.TH "RunGui" 3 "Fri Jun 12 2015" "Version 0.0.1" "RIDE" \" -*- nroff -*-
.ad l
.nh
.SH NAME
RunGui \- 
.SH SYNOPSIS
.br
.PP
.PP
\fC#include <RunGui\&.h>\fP
.PP
Inherits QWidget\&.
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "\fBRunGui\fP (QWidget *parent=0)"
.br
.ti -1c
.RI "void \fBinitBtns\fP ()"
.br
.ti -1c
.RI "void \fBreset\fP ()"
.br
.ti -1c
.RI "void \fBsetLaunchFilePathStrPtr\fP (QString *\fBlaunchFilePathStrPtr\fP)"
.br
.ti -1c
.RI "QString * \fBgetLaunchFilePathStrPtr\fP ()"
.br
.ti -1c
.RI "void \fBswapNextPage\fP ()"
.br
.ti -1c
.RI "void \fBswapBackPage\fP ()"
.br
.ti -1c
.RI "void \fBloadPage_1\fP ()"
.br
.ti -1c
.RI "void \fBunloadPage_1\fP ()"
.br
.ti -1c
.RI "void \fBloadPage_2\fP ()"
.br
.ti -1c
.RI "void \fBunloadPage_2\fP ()"
.br
.ti -1c
.RI "void \fBloadPage_3\fP ()"
.br
.ti -1c
.RI "void \fBunloadPage_3\fP ()"
.br
.ti -1c
.RI "QString * \fBtoString\fP ()"
.br
.ti -1c
.RI "\fB~RunGui\fP ()"
.br
.in -1c
.SS "Private Types"

.in +1c
.ti -1c
.RI "enum \fBPage\fP { \fBPAGE_ONE\fP, \fBPAGE_TWO\fP, \fBPAGE_THREE\fP }"
.br
.in -1c
.SS "Private Slots"

.in +1c
.ti -1c
.RI "void \fBhandleBackBtnSlot\fP ()"
.br
.ti -1c
.RI "void \fBhandleNextBtnSlot\fP ()"
.br
.ti -1c
.RI "void \fBhandleFinishBtnSlot\fP ()"
.br
.ti -1c
.RI "void \fBhandleHelpBntSlot\fP ()"
.br
.ti -1c
.RI "void \fBhandleCancelBtnSlot\fP ()"
.br
.in -1c
.SS "Private Attributes"

.in +1c
.ti -1c
.RI "\fBPage\fP \fBcurrentPage\fP"
.br
.ti -1c
.RI "QString * \fBlaunchFilePathStrPtr\fP"
.br
.ti -1c
.RI "QPushButton * \fBbackBtn\fP"
.br
.ti -1c
.RI "QPushButton * \fBnextBtn\fP"
.br
.ti -1c
.RI "QPushButton * \fBfinishBtn\fP"
.br
.ti -1c
.RI "QPushButton * \fBhelpBtn\fP"
.br
.ti -1c
.RI "QPushButton * \fBcancelBtn\fP"
.br
.ti -1c
.RI "QGridLayout * \fBouterLayoutPtr\fP"
.br
.ti -1c
.RI "QHBoxLayout * \fBbuttonLayout\fP"
.br
.ti -1c
.RI "\fBRunPage_1\fP * \fBrunPage_1Ptr\fP"
.br
.ti -1c
.RI "\fBRunPage_2\fP * \fBrunPage_2Ptr\fP"
.br
.ti -1c
.RI "\fBRunPage_3\fP * \fBrunPage_3Ptr\fP"
.br
.in -1c
.SH "Detailed Description"
.PP 
Definition at line 27 of file RunGui\&.h\&.
.SH "Member Enumeration Documentation"
.PP 
.SS "enum \fBRunGui::Page\fP\fC [private]\fP"

.PP
\fBEnumerator\fP
.in +1c
.TP
\fB\fIPAGE_ONE \fP\fP
.TP
\fB\fIPAGE_TWO \fP\fP
.TP
\fB\fIPAGE_THREE \fP\fP
.PP
Definition at line 32 of file RunGui\&.h\&.
.SH "Constructor & Destructor Documentation"
.PP 
.SS "RunGui::RunGui (QWidget *parent = \fC0\fP)"

.PP
Definition at line 4 of file RunGui\&.cpp\&.
.SS "RunGui::~RunGui ()"

.PP
Definition at line 228 of file RunGui\&.cpp\&.
.SH "Member Function Documentation"
.PP 
.SS "QString * RunGui::getLaunchFilePathStrPtr ()"

.PP
Definition at line 106 of file RunGui\&.cpp\&.
.SS "void RunGui::handleBackBtnSlot ()\fC [private]\fP, \fC [slot]\fP"

.PP
Definition at line 22 of file RunGui\&.cpp\&.
.SS "void RunGui::handleCancelBtnSlot ()\fC [private]\fP, \fC [slot]\fP"

.PP
Definition at line 57 of file RunGui\&.cpp\&.
.SS "void RunGui::handleFinishBtnSlot ()\fC [private]\fP, \fC [slot]\fP"

.PP
Definition at line 34 of file RunGui\&.cpp\&.
.SS "void RunGui::handleHelpBntSlot ()\fC [private]\fP, \fC [slot]\fP"

.PP
Definition at line 51 of file RunGui\&.cpp\&.
.SS "void RunGui::handleNextBtnSlot ()\fC [private]\fP, \fC [slot]\fP"

.PP
Definition at line 28 of file RunGui\&.cpp\&.
.SS "void RunGui::initBtns ()"

.PP
Definition at line 63 of file RunGui\&.cpp\&.
.SS "void RunGui::loadPage_1 ()"

.PP
Definition at line 172 of file RunGui\&.cpp\&.
.SS "void RunGui::loadPage_2 ()"

.PP
Definition at line 188 of file RunGui\&.cpp\&.
.SS "void RunGui::loadPage_3 ()"

.PP
Definition at line 204 of file RunGui\&.cpp\&.
.SS "void RunGui::reset ()"

.PP
Definition at line 91 of file RunGui\&.cpp\&.
.SS "void RunGui::setLaunchFilePathStrPtr (QString *launchFilePathStrPtr)"

.PP
Definition at line 100 of file RunGui\&.cpp\&.
.SS "void RunGui::swapBackPage ()"

.PP
Definition at line 112 of file RunGui\&.cpp\&.
.SS "void RunGui::swapNextPage ()"

.PP
Definition at line 136 of file RunGui\&.cpp\&.
.SS "QString * RunGui::toString ()"

.PP
Definition at line 220 of file RunGui\&.cpp\&.
.SS "void RunGui::unloadPage_1 ()"

.PP
Definition at line 180 of file RunGui\&.cpp\&.
.SS "void RunGui::unloadPage_2 ()"

.PP
Definition at line 196 of file RunGui\&.cpp\&.
.SS "void RunGui::unloadPage_3 ()"

.PP
Definition at line 212 of file RunGui\&.cpp\&.
.SH "Member Data Documentation"
.PP 
.SS "QPushButton* RunGui::backBtn\fC [private]\fP"

.PP
Definition at line 43 of file RunGui\&.h\&.
.SS "QHBoxLayout* RunGui::buttonLayout\fC [private]\fP"

.PP
Definition at line 50 of file RunGui\&.h\&.
.SS "QPushButton* RunGui::cancelBtn\fC [private]\fP"

.PP
Definition at line 47 of file RunGui\&.h\&.
.SS "\fBPage\fP RunGui::currentPage\fC [private]\fP"

.PP
Definition at line 39 of file RunGui\&.h\&.
.SS "QPushButton* RunGui::finishBtn\fC [private]\fP"

.PP
Definition at line 45 of file RunGui\&.h\&.
.SS "QPushButton* RunGui::helpBtn\fC [private]\fP"

.PP
Definition at line 46 of file RunGui\&.h\&.
.SS "QString* RunGui::launchFilePathStrPtr\fC [private]\fP"

.PP
Definition at line 41 of file RunGui\&.h\&.
.SS "QPushButton* RunGui::nextBtn\fC [private]\fP"

.PP
Definition at line 44 of file RunGui\&.h\&.
.SS "QGridLayout* RunGui::outerLayoutPtr\fC [private]\fP"

.PP
Definition at line 49 of file RunGui\&.h\&.
.SS "\fBRunPage_1\fP* RunGui::runPage_1Ptr\fC [private]\fP"

.PP
Definition at line 52 of file RunGui\&.h\&.
.SS "\fBRunPage_2\fP* RunGui::runPage_2Ptr\fC [private]\fP"

.PP
Definition at line 53 of file RunGui\&.h\&.
.SS "\fBRunPage_3\fP* RunGui::runPage_3Ptr\fC [private]\fP"

.PP
Definition at line 54 of file RunGui\&.h\&.

.SH "Author"
.PP 
Generated automatically by Doxygen for RIDE from the source code\&.


src/RunGui.cpp
#include "RunGui.h"


RunGui::RunGui(QWidget* parent) : QWidget(parent)
{
    this->setWindowIcon(QIcon("/home/james/NetBeansProjects/ride/images/project2.jpg") );
    this->setWindowTitle("Ride");
    
    outerLayoutPtr = new QGridLayout();
    runPage_1Ptr = new RunPage_1();
    runPage_2Ptr = new RunPage_2();
    runPage_3Ptr = new RunPage_3();
    
    currentPage = PAGE_ONE;

    initBtns();
    
    this->setLayout(outerLayoutPtr);
}


void RunGui::handleBackBtnSlot()
{
    swapBackPage();
}


void RunGui::handleNextBtnSlot()
{
    swapNextPage();
}


void RunGui::handleFinishBtnSlot()
{
    if(*runPage_1Ptr->getRunOptionPtr() == "Launch File")
    {
        setLaunchFilePathStrPtr(runPage_2Ptr->getLaunchFileStrPtr() );
    }
    else if (*runPage_1Ptr->getRunOptionPtr() == "Individual Nodes")
    {
        cerr << "setting run node stuff has not been done" << endl;
    }
    else
    {
        cerr << "Invalid option at RunGui::handleFinishBtnSlot()" << endl;
    }
}


void RunGui::handleHelpBntSlot()
{
    ;
}


void RunGui::handleCancelBtnSlot()
{
    this->close();
}


void RunGui::initBtns()
{
    backBtn = new QPushButton("< Back", this);
    backBtn->setEnabled(false);
    nextBtn = new QPushButton("Next >", this);
    finishBtn = new QPushButton("Finish", this);
    finishBtn->setEnabled(false);
    helpBtn = new QPushButton("Help", this);
    cancelBtn = new QPushButton("Cancel", this);
    
    buttonLayout = new QHBoxLayout();
    buttonLayout->addWidget(backBtn);
    buttonLayout->addWidget(nextBtn);
    buttonLayout->addWidget(finishBtn);
    buttonLayout->addWidget(helpBtn);
    buttonLayout->addWidget(cancelBtn);
    
    connect(backBtn, SIGNAL(released() ), this, SLOT(handleBackBtnSlot() ) );
    connect(nextBtn, SIGNAL(released() ), this, SLOT(handleNextBtnSlot() ) );
    connect(finishBtn, SIGNAL(released() ), this, SLOT(handleFinishBtnSlot() ) );
    connect(helpBtn, SIGNAL(released() ), this, SLOT(handleHelpBntSlot() ) );
    connect(cancelBtn, SIGNAL(released() ), this, SLOT(handleCancelBtnSlot() ) );
    
    outerLayoutPtr->addWidget(runPage_1Ptr, 0, 0);
    outerLayoutPtr->addLayout(buttonLayout, 1, 0, Qt::AlignBottom);
}


void RunGui::reset()
{
    if(currentPage != PAGE_ONE)
    {
        swapBackPage();    
    }
}


void RunGui::setLaunchFilePathStrPtr(QString* launchFilePathStrPtr)
{
    this->launchFilePathStrPtr = launchFilePathStrPtr;
}


QString* RunGui::getLaunchFilePathStrPtr()
{
    return launchFilePathStrPtr;
}


void RunGui::swapBackPage()
{
    switch(currentPage)
    {
        case PAGE_TWO:
            unloadPage_2();
            loadPage_1();
            currentPage = PAGE_ONE;
            nextBtn->setEnabled(true);
            finishBtn->setEnabled(false);
            break;
        case PAGE_THREE:
            unloadPage_3();
            loadPage_1();
            currentPage = PAGE_ONE;
            nextBtn->setEnabled(true);
            finishBtn->setEnabled(false);
            break;
        default:
            cerr << "ERROR in switch at: NewFileGui::swapBackPage()" << endl;
    }
}


void RunGui::swapNextPage()
{
    switch(currentPage)
    {
        case PAGE_ONE:
            unloadPage_1();
            
            if(*runPage_1Ptr->getRunOptionPtr() == "Launch File")
            {
                loadPage_2();
                currentPage = PAGE_TWO;
                
                backBtn->setEnabled(true);
                nextBtn->setEnabled(false);
                finishBtn->setEnabled(true);
            }
            else if (*runPage_1Ptr->getRunOptionPtr() == "Individual Nodes")
            {
                loadPage_3();
                currentPage = PAGE_THREE;
                
                backBtn->setEnabled(true);
                nextBtn->setEnabled(false);
                finishBtn->setEnabled(true);
            }
            else
            {
                cerr << "Error at RunGui::SwapNextPage(); " << runPage_1Ptr->getRunOptionPtr()->toStdString() << endl;
            }
            break;
        default:
            cerr << "ERROR in switch at: NewFileGui::swapNextPage()" << endl;
    }
}


void RunGui::loadPage_1()
{
    outerLayoutPtr->addWidget(runPage_1Ptr, 0, 0);
    runPage_1Ptr->setVisible(true);
    runPage_1Ptr->setEnabled(true);
}


void RunGui::unloadPage_1()
{
    outerLayoutPtr->removeWidget(runPage_1Ptr);
    runPage_1Ptr->setVisible(false);
    runPage_1Ptr->setEnabled(false);
}


void RunGui::loadPage_2()
{
    outerLayoutPtr->addWidget(runPage_2Ptr, 0, 0);
    runPage_2Ptr->setVisible(true);
    runPage_2Ptr->setEnabled(true);
}


void RunGui::unloadPage_2()
{
    outerLayoutPtr->removeWidget(runPage_2Ptr);
    runPage_2Ptr->setVisible(false);
    runPage_2Ptr->setEnabled(false);
}


void RunGui::loadPage_3()
{
    outerLayoutPtr->addWidget(runPage_3Ptr, 0, 0);
    runPage_3Ptr->setVisible(true);
    runPage_3Ptr->setEnabled(true);
}


void RunGui::unloadPage_3()
{
    outerLayoutPtr->removeWidget(runPage_3Ptr);
    runPage_3Ptr->setVisible(false);
    runPage_3Ptr->setEnabled(false);
}


RunPage_1* RunGui::getRunPage_1Ptr()
{
    return runPage_1Ptr;
}


RunPage_2* RunGui::getRunPage_2Ptr()
{
    return runPage_2Ptr;
}


RunPage_3* RunGui::getRunPage_3Ptr()
{
    return runPage_3Ptr;
}


QString* RunGui::toString()
{
    QString* tmp = new QString("RunGui");
    
    return tmp;
}


RunGui::~RunGui()
{
    ;
}